src/util/Rectangle.cpp

```diff
@@ -4,18 +4,20 @@
 
 #include "Range.h"
 
-Rectangle::Rectangle() = default;
+template <class T>
+Rectangle<T>::Rectangle() = default;
 
-Rectangle::Rectangle(double x, double y, double width, double height): x(x), y(y), width(width), height(height) {}
+template <class T>
+Rectangle<T>::Rectangle(T x, T y, T width, T height): x(x), y(y), width(width), height(height) {}
 
-Rectangle::Rectangle(const Range& rect):
+template <class T>
+Rectangle<T>::Rectangle(const Range& rect):
         x(rect.getX()), y(rect.getY()), width(rect.getWidth()), height(rect.getHeight()) {}
 
-Rectangle::~Rectangle() = default;
-
-auto Rectangle::intersects(const Rectangle& other, Rectangle* dest) const -> bool {
-    double destX = NAN, destY = NAN;
-    double destW = NAN, destH = NAN;
+template <class T>
+auto Rectangle<T>::intersects(const Rectangle<T>& other, Rectangle<T>* dest) const -> bool {
+    T destX = NAN, destY = NAN;
+    T destW = NAN, destH = NAN;
```

Comment:
These declarations should be removed and the `destX = std::max(...)` lines should become declarations. `NAN` is not a valid value for any non-floating point number.